src/App.tsx

```diff
@@ -15,37 +17,105 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SetByReverse = true;
+
+enum SortedFields {
+  SetById,
+  SetByAlphabet,
+  SetByLength,
+  SetByDanger,
+}
+
+function getSortedGoods(
+  goods: string[],
+  { sortField, reversed }: { sortField: SortedFields; reversed: boolean },
+) {
+  const preparedGoods = [...goods];
+
+  preparedGoods.sort((good1, good2) => {
+    switch (sortField) {
+      case SortedFields.SetById:
+      case SortedFields.SetByAlphabet:
+        return good1.localeCompare(good2);
+      case SortedFields.SetByLength:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (reversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setsortField] = useState(SortedFields.SetByDanger);
+  const [reversed, setReversed] = useState(false);
+
+  const sortedGoods = getSortedGoods(goodsFromServer, { sortField, reversed });
+
+  const onReset = () => {
+    setReversed(false);
+    setsortField(SortedFields.SetByDanger);
+  };
+
+  const onReverse = () => {
+    setReversed(!reversed);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => setsortField(SortedFields.SetByAlphabet)}
+          type="button"
+          className={cn(
+            'button is-info',
+
+            { 'is-light': sortField !== SortedFields.SetByAlphabet },
+          )}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => setsortField(SortedFields.SetByLength)}
+          type="button"
+          className={cn(
+            'button is-success',
+
+            { 'is-light': sortField !== SortedFields.SetByLength },
+          )}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={onReverse}
+          type="button"
+          className={cn(
+            'button is-warning',
+
+            { 'is-light': reversed !== SetByReverse },
```

Comment:
The comparison 'reversed !== SetByReverse' will always return true since 'SetByReverse' is a constant set to 'true' and 'reversed' is a boolean state that can be either true or false. This logic does not seem to serve the intended purpose. If you want to toggle the button's appearance based on the 'reversed' state, simply use '{ 'is-light': !reversed }'.